frame/staking/src/tests.rs

```diff
@@ -5560,3 +5561,76 @@ fn reducing_history_depth_without_migration() {
 		HistoryDepth::set(original_history_depth);
 	});
 }
+
+#[test]
+fn reducing_max_unlocking_chunks_abrupt() {
+	// Concern is on validators only
+	// By Default 11, 10 are stash and ctrl and 21,20
+	ExtBuilder::default().build_and_execute(|| {
+		// given a staker at era=10 and MaxUnlockChunks set to 2
+		MaxUnlockingChunks::set(2);
+		start_active_era(10);
+		let expected_claimed_rewards: BoundedVec<_, _> =
+			(0..10).collect::<Vec<_>>().try_into().unwrap();
+
+		assert_ok!(Staking::bond(RuntimeOrigin::signed(3), 4, 300, RewardDestination::Staked));
+		assert_eq!(
```

Comment:
A rust note: since we only care about `unlokcing` here, an alternative syntax is 

```
assert!(matches!(x, StakingLedher { unlocking: bounded_vec![..], ..));
```